tls/s2n_cipher_suites.c

```diff
@@ -1215,7 +1215,7 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
             }
 
             /* If connection is for SSLv3, use SSLv3 version of suites */
-            if (conn->client_protocol_version == S2N_SSLv3) {
+            if (conn->actual_protocol_version == S2N_SSLv3) {
```

Comment:
This is very similar to https://github.com/aws/s2n-tls/pull/3522#discussion_r984082682

Basically, the server could check client_protocol_version here instead of actual_protocol_version because the server would never negotiate SSLv3 unless that was the client version. But in my test, I force the server to do SSLv3 regardless of client version.